src/baal/active/dataset.py

```diff
@@ -94,8 +103,12 @@ def pool(self) -> torchdata.Dataset:
         """Returns a new Dataset made from unlabelled samples"""
         pool_dataset = copy(self._dataset)
         # TODO Handle target transform as well.
-        if hasattr(pool_dataset, 'transform') and self.eval_transform is not None:
-            pool_dataset.transform = self.eval_transform
+
+        for attr, new_val in self.pool_specifics.items():
+            if hasattr(pool_dataset, attr):
+                setattr(pool_dataset, attr, new_val)
+            else:
+                raise ValueError(f"{pool_dataset} doesn't have {attr}")
```

Comment:
I was wondering if it would be worth adding to the `docstring` the fact that an exception is raised. This is a property method though, so I'm not sure.

Reply:
Should we make this validation in the constructor so that the property is exception free?